mypy/semanal.py

```diff
@@ -719,9 +719,54 @@ def analyze_class_body(self, defn: ClassDef) -> Iterator[bool]:
                 yield True
                 self.calculate_abstract_status(defn.info)
                 self.setup_type_promotion(defn)
-
+                self.apply_class_plugin_hooks(defn)
                 self.leave_class()
 
+    def apply_class_plugin_hooks(self, defn: ClassDef) -> None:
+        """Apply a plugin hook that may infer a more precise definition for a class."""
+
+        def get_fullname(expr: Expression) -> Optional[str]:
+            # We support foo.bar(...), foo.bar, and bar.
+            if isinstance(expr, CallExpr):
+                if isinstance(expr.callee, RefExpr):
+                    return expr.callee.fullname
```

Comment:
``RefExpr`` can be both ``NameExpr`` and ``MemberExpr``. Why do you have the same code for both here, while below you need special casing?

Reply:
I should probably just recurse.